app/controllers/spots_controller.rb

```diff
@@ -0,0 +1,29 @@
+class SpotsController < ApplicationController
+  def index
+    @spots = Spot.all
+  end
+
+  def new
+    @spot = Spot.new
+  end
+
+  def create
+    @spot = Spot.new(spot_params)
+    if @spot.save
+      flash[:notice] = 'Spot added.'
+      redirect_to spots_path
+    else
+      flash[:alert] = @spot.errors.full_messages
```

Comment:
You are printing error messages in your views/spots/new.html.erb. I wouldn't worry about putting that information into the hash. You could make this message more general: "Spot was not saved"